node/bft/src/primary.rs

```diff
@@ -488,90 +488,114 @@ impl<N: Network> Primary<N> {
+                        // Ensure the checksum matches. If not, skip the transaction.
+                        if !matches!(transaction.to_checksum::<N>(), Ok(transaction_checksum) if transaction_checksum == checksum )
+                        {
+                            trace!("Proposing - Skipping transaction '{}' - Checksum mismatch", fmt_id(transaction_id));
+                            continue;
+                        }
+
+                        // Deserialize the transaction. If the transaction exceeds the maximum size, then return an error.
+                        let transaction = spawn_blocking!({
+                            match transaction {
+                                Data::Object(transaction) => Ok(transaction),
+                                Data::Buffer(bytes) => {
+                                    Ok(Transaction::<N>::read_le(&mut bytes.take(N::MAX_TRANSACTION_SIZE as u64))?)
                                 }
                             }
-                            // Check if the solution is still valid.
-                            if let Err(e) = self.ledger.check_solution_basic(solution_id, solution).await {
-                                trace!("Proposing - Skipping solution '{}' - {e}", fmt_id(solution_id));
-                                continue 'inner;
-                            }
+                        })?;
+
+                        // Check if the transaction is still valid.
+                        // TODO: check if clone is cheap, otherwise fix.
+                        if let Err(e) = self.ledger.check_transaction_basic(transaction_id, transaction.clone()).await {
+                            trace!("Proposing - Skipping transaction '{}' - {e}", fmt_id(transaction_id));
+                            continue;
                         }
-                        (
-                            TransmissionID::Transaction(transaction_id, checksum),
-                            Transmission::Transaction(transaction),
-                        ) => {
-                            // Ensure the checksum matches.
-                            match transaction.to_checksum::<N>() {
-                                Ok(transaction_checksum) if transaction_checksum == checksum => (),
+
+                        // Ensure the transaction doesn't bring the proposal above the spend limit.
+                        //
+                        // Validators start following the new consensus rules early, in order to guarantee a smooth transition.
+                        // The activation height is offset by the maximum number of blocks that can be produced within gc to
+                        // ensure honest validators can't mismatch due to differing views of the dag.
+                        let block_height =
+                            self.ledger.latest_block_height() + 1 + (BatchHeader::<N>::MAX_GC_ROUNDS as u32 / 2);
```

Comment:
nit: use `saturating_*` for adds and divs

Reply:
Added in [17ddf90](https://github.com/ProvableHQ/snarkOS/pull/3471/commits/17ddf908b06dae6aac0f812a6df7cc8338fecc07).